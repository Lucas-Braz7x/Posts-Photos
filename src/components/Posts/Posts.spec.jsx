import { render, screen } from '@testing-library/react';
import {Posts} from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'titulo 1',
      body: 'body 1',
      cover: 'img/img1.png'
    },
    {
      id: 1,
      title: 'titulo 2',
      body: 'body 2',
      cover: 'img/img2.png'
    },
    {
      id: 3,
      title: 'titulo 3',
      body: 'body 3',
      cover: 'img/img3.png'
    }
  ]
} 

describe('<Posts />', () =>{
  it('should render posts', () =>{
    render(<Posts posts={props.posts} />);

    expect(screen.getAllByRole('heading', {name: /titulo/}))
    .toHaveLength(3);
    expect(screen.getAllByRole('img', {name: /titulo/i}))
    .toHaveLength(3);
    expect(screen.getAllByText(/body/i))
    .toHaveLength(3);
    expect(screen.getByRole('img', {name: /titulo 3/i}))
    .toHaveAttribute('src', 'img/img3.png');
  });
  it('should match snapshot', () =>{
    const {container} = render(<Posts posts={props.posts} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});