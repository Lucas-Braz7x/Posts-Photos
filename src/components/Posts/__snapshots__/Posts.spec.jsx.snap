// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="titulo 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        titulo 1
      </h1>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="titulo 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        titulo 2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="titulo 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        titulo 3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
